src/Generator/PropertyConditionsGenerator.php

```diff
@@ -112,16 +112,10 @@ private function isAllowedAttribute(PropertyMapping $propertyMapping): ?Expr
 
         $variableRegistry = $mapperMetadata->getVariableRegistry();
 
-        /** Create expression on how to read the value from the source */
-        $sourcePropertyAccessor = new Expr\Assign(
-            $variableRegistry->getFieldValueVariable($propertyMapping),
-            $propertyMapping->readAccessor->getExpression($variableRegistry->getSourceInput())
-        );
-
         return new Expr\StaticCall(new Name\FullyQualified(MapperContext::class), 'isAllowedAttribute', [
             new Arg($variableRegistry->getContext()),
             new Arg(new Scalar\String_($propertyMapping->property)),
-            new Arg($sourcePropertyAccessor),
+            new Arg($propertyMapping->readAccessor->getIsNullExpression($variableRegistry->getSourceInput())),
```

Comment:
this might change the generated code: maybe the phpdoc of this method should be updated?